python_modules/dagster/dagster/_core/definitions/asset_condition_evaluation_context.py

```diff
@@ -38,6 +52,55 @@ class RootAssetConditionEvaluationContext:
     evaluation_results_by_key: Mapping[AssetKey, "AssetConditionEvaluation"]
     expected_data_time_mapping: Mapping[AssetKey, Optional[datetime.datetime]]
 
+    root_ref: Optional["AssetConditionEvaluationContext"] = None
+
+    @staticmethod
+    def create(
```

Comment:
I think this was just a habit picked up from namedtuple land -- I can update this

Reply:
ah actually it seems like it's a bit of a pain to override the __init__ function because you'd have to manually to all the `self.x = y` stuff yourself. in this case, feels like this is a reasonable pattern given the args don't map 1-1 here

Reply:
you can't use `super().__init__(`?

anyway main suggestion is just to add docstring explaining why a staticmethod makes sense